<launch>

    <param name="lidar_frame" value="laser"/>
    <param name="lidar_topic" value="/points_raw"/>

    <param name="tf_x"     value="0.37" />
    <param name="tf_y"     value="0" />
    <param name="tf_z"     value="0.275" />
    <param name="tf_roll"  value="0" />
    <param name="tf_pitch" value="0" />
    <param name="tf_yaw"   value="-1.57" />

    <arg name="method_type" default="0" /> <!-- pcl_generic=0, ndt_cpu=1, ndt_gpu=2, pcl_openmp=3 -->
    <arg name="use_gnss" default="0" />
    <arg name="use_odom" default="false" />
    <arg name="use_imu" default="false" />
    <arg name="imu_upside_down" default="false" />
    <arg name="imu_topic" default="/imu_raw" />
    <arg name="queue_size" default="1" />
    <arg name="offset" default="linear" />
    <arg name="get_height" default="false" />
    <arg name="use_local_transform" default="false" />
    <arg name="sync" default="false" />
    <arg name="output_log_data" default="false" />
    <arg name="output_tf_frame_id" default="base_link"/>
    <arg name="gnss_reinit_fitness" default="500.0" />
    <arg name="voxel_leaf_size" default="1.5" />

    <node pkg="htc_ndt_mapping" type="yunle_ndt_mapping" name="yunle_ndt_mapping" output="screen">
        <!-- <param name="lidar_frame" value="$(arg lidar_frame)" />
        <param name="lidar_topic" value="$(arg lidar_topic)" /> -->

        <param name="method_type" value="$(arg method_type)" /> <!-- pcl_generic=0, pcl_cpu=1, pcl_gpu=2, pcl_openmp=3 -->
        <param name="use_imu" value="false" />
        <param name="use_odom" value="false" />
        <param name="imu_upside_down" value="false" />
        <param name="imu_topic" value="$(arg imu_topic)" />
        <param name="incremental_voxel_update" value="false" />

        <param name="ndt_res" value="0.5" />
        <param name="step_size" value="0.1" />
        <param name="trans_eps" value="0.01" />
        <param name="max_iter" value="30" />
        <param name="voxel_leaf_size" value="$(arg voxel_leaf_size)" />
        <param name="min_scan_range" value="0.4" />
        <param name="max_scan_range" value="80" />
        <param name="min_add_scan_shift" value="1.0" />
        <param name="is_filter_before_add_to_map" value="true" />
        <param name="voxel_size_filter_before_add_to_map" value="0.5" />
    </node>

</launch>